docs/motivation-and-use-cases.md

```diff
@@ -0,0 +1,138 @@
+
+
+## Home directories with ipfs-cluster (first reported here https://github.com/ipfs/ipfs-cluster/issues/3)
+
+Actors:
+	file system users
+	system admin
+
+Description:  An admin wishing to host a distributed filesystem on ipfs-cluster sets up a few ipfs-cluster machines on dedicated nodes on a WAN.  Nodes in the same LAN or same region are organized in a sub cluster.  The admin begins by adding existing home directories to ipfs-cluster.  This includes adding unix style authentication information to different directories and files so that only those with the right permissions can access files.  After set up is complete, users begin accessing their files, reading, writing and sharing amongst themselves.  When users get started with the new file system they add a new ipfs-cluster peer located on their local machine or workstation to the cluster, this way they can contribute to storing and distributing user data while accessing files.  Throughout the life-time of the distributed file-system’s use the admin administers the underlying ipfs-cluster with a web-based management console.  Clients can provide quotas across a couple of metrics (ex: bandwidth and storage) so that ipfs-cluster doesn’t hog all their resources.
+
+Implied ipfs-cluster requirements:
+
+- ipfs-cluster provides key management, allow for per-user keys and implements built in encryption for robust, out-of-the-box security. 
+- ipfs-cluster’s replication strategies are sophisticated enough to achieve excellent availability, this may include live, feedback-based replication strategies that take into consideration the popularity and age of files
+- ipfs-cluster exposes a management console that allows the setting of authentication and encryption metadata, and setting policies for ipfs-cluster as a unit and the sub clusters of which it consists.  Policy parameters include replication schemes and allocation schemes based on storage space, location, bandwidth and subcluster membership.  
+- ipfs-cluster implements complex allocation strategies and supports the policies that outline these strategies, as well as live updates of these policies
+- ipfs-cluster provides configurable load balancing across ipfs-cluster nodes (WAN cluster and LAN subclusters have different balancers), that can handle frequent changes to peersets
+- ipfs-clusters are easy to join and leave without having to think too much about set-up or generating errors
+- ipfs-cluster allows for retrieval of data across sub clusters that are not necessarily connected, except as two subtrees of the same larger cluster.
+ipfs-cluster allows individual peers to specify resource constraints 
```

Comment:
This is closely related to files api and unixfs/fuse right? It's like mounting the ipfs fuse filesystem in /home and using cluster to make sure that the contents on it are propagated to all other machines/users.

Reply:
I agree, when I pass through and list features that cluster requires I'll add things along these lines.